src/anndata/_core/sparse_dataset.py

```diff
@@ -181,7 +181,11 @@ def _get_sliceXslice(self, row: slice, col: slice) -> ss.csr_matrix:
                     self._get_contiguous_compressed_slice(row), shape=out_shape
                 )
             return self._get_arrayXslice(np.arange(*row.indices(self.shape[0])), col)
-        return super()._get_sliceXslice(row, col)
+
+        # This is better than super()._get_sliceXslice,
+        # as that loads the complete .data and .indices arrays into memory
+        # TODO: Can probably still be more efficient
+        return self[row, :][:, col]
```

Comment:
I can't re-write the function in a suggestion but I would probably change the `elif` block above to be an `else` block and then `res = ss.csr_matrix(...` which could be indexed by `col` afterwards if `out_shape[1] < shape[1]`

Reply:
I think so. Let’s see if what I think you meant works

Reply:
Seems like it!